### DISCUSSION TITLE
Is it possible to compare tags?

### DISCUSSION THREAD

**Discussion by van-de-bugger** [quodlibet/quodlibet#4338] [Q&A] [answered]
Hi all,

Is it possible to compare tags in search expressions and/or tag patterns? I want to show year of an album, but only if it differs from the original year:

    <~originalyear> · <album> <~year = ~originalyear|| · <~year>>

but without any success. It looks like the condition is always false, so the year is always shown. I tried to use numeric comparison

    <~originalyear> · <album> <#(~#year = ~#originalyear)|| · <~year>>

but it does not work also. Comparing with a constant does work:

    <~originalyear> · <album> <~year ="2005"|| · <~year>>

but it is not what I want.

**Comment by fmaindrault** [comment, marked as answer]
In case you're still looking for this, try using:
`<~originalyear> · <album> <~~year~~originalyear = /(.+) - \1/|| · <~year>>`

It's the only way to do it that I could find while trying something similar.
It concatenates the 2 fields (with " - " as separator), then checks with a regex that the result is twice the same thing with ' - ' in between.

For non numeric fields (like artist and albumartist), you should probably use `/^(.+) - \1$/` for the regex to ensure the whole value is compared. There could still be false positive in weird cases, but I don't think they can happen in real life situation.

**Reply by van-de-bugger** [reply]
Hi,

Thanks for the workaround, it works.
